### DISCUSSION TITLE
How to get metrics from inference?

### DISCUSSION THREAD

**Discussion by elsxnh** [ultralytics/ultralytics#4331] [Q&A]
Hello, I'm currently trying to compare the performance metrics from the default yolov8 model and the custom trained yolov8 model that I did, so I'm trying to run predict on some images. Is there any way to get the performance metrics like F1, precision, recall and mAP for these tested images? If so, how do I do it? Since, so far, I notice that these are by default obtained only when we are training our own custom model. 

Thank you in advance!

**Comment by sketcch** [comment]
If these images used for inference are labeled, you can use `val.py` to validate on these images and get the metrics.

**Reply by glenn-jocher** [reply]
@HoshiMeisa hello! To compare performance metrics like F1, precision, recall, and mAP between the default YOLOv8 model and your custom-trained model, you can indeed use the `val` mode with labeled images. This will provide you with the desired metrics for your test set. Just ensure that your test dataset is properly set up with annotations in a format that YOLOv8 can understand, and you're good to go. Happy validating! 😊📊

**Comment by hannaliavoshka** [comment]
Use validation mode instead of prediction mode:
`model.val()`
As mentioned above, this will work if the data is labeled and has .yaml description.
https://docs.ultralytics.com/modes/val/
https://docs.ultralytics.com/datasets/detect/

**Reply by glenn-jocher** [reply]
@hannaliavoshka hello! To compare performance metrics such as F1, precision, recall, and mAP between your default and custom-trained YOLOv8 models, you should use the validation mode (`model.val()`) rather than the prediction mode. This requires your test images to be labeled and accompanied by a dataset configuration file in YAML format. The validation mode will provide you with the desired metrics, assuming your dataset is properly set up. For more details on how to use the validation mode and prepare your dataset, please refer to the Ultralytics Docs on validation and datasets for detection. 📊🔍
